source/common/router/router.cc

```diff
@@ -1038,7 +1038,7 @@ void Filter::onStreamMaxDurationReached(UpstreamRequest& upstream_request) {
       StreamInfo::ResponseFlag::UpstreamMaxStreamDurationReached);
   // sendLocalReply may instead reset the stream if downstream_response_started_ is true.
   callbacks_->sendLocalReply(
-      Http::Code::RequestTimeout, "upstream max stream duration reached", modify_headers_,
+      Http::Code::GatewayTimeout, "upstream max stream duration reached", modify_headers_,
```

Comment:
I think stream max duration can be reached with the incomplete downstream request, right? If so then I think we do need to check here if the downstream request is complete to distinguish between 408 and 504.

Reply:
Thank you for the comment!

To make it clear, do you mean the situation that downstream headers are complete (for upstream routing)
but the downstream request body is not complete?

I am not sure if we want to fix all cases or some fix is better than nothing.

FYI:  My next https://github.com/envoyproxy/envoy/pull/20026 starts to consider this corner case but the implementation is built upon my poor knowledge on HCM. I am OK to hold and merge this to that one once that one receives more feedback and polished

Reply:
Let's close this: this case should be covered by https://github.com/envoyproxy/envoy/pull/20026

Reply:
Thank @yanavlasov for the feedback!